prism/src/main/java/org/apache/falcon/resource/extensions/ExtensionManager.java

```diff
@@ -420,6 +421,22 @@ public String getExtensionDescription(
         }
     }
 
+    @GET
+    @Path("detail/{extension-name}")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response getDetail(@PathParam("extension-name") String extensionName){
+        checkIfExtensionServiceIsEnabled();
+        validateExtensionName(extensionName);
+        JSONObject result = new JSONObject();
+        try {
+            result.put(EXTENSION_DETAIL,buildDetailResult(extensionName));
```

Comment:
why JSON inside a JSON? Can you send the json object from buildDetailResult directly?